service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/JobProperties.java

```diff
@@ -303,14 +323,16 @@ JobPropertiesParser toJobPropertiesParser()
      *                              Including a SAS token is dependent on the StorageAuthenticationType
      * @param outputBlobContainerUri URI to a blob container. This is used to output the status of the job and the results.
      *                               Including a SAS token is dependent on the StorageAuthenticationType
+     * @param identity the managed identity used to access the storage account for import jobs.
      * @return An instance of JobProperties
      */
     public static JobProperties createForImportJob(
```

Comment:
Do customers call this method? If so, this looks like a breaking change

Reply:
Why would it be a breaking change? It is just adding a new optional property. Customers can either set it or leave it alone.

Reply:
Java doesn't have optional parameters, so they can't leave it alone. If someone was calling this method with 2 parameters before, it wouldn't compile once you add a third parameter

Reply:
Typically when you add an optional parameter to a java method, you need to create a new method like

```java
public void myMethod(int arg1, int arg2)
{
    myMethod(arg1, arg2, 0)
}

public void myMethod(int arg1, int arg2, int arg3)
{

}
```

Reply:
Ahh I think I get it now. Some how your comment did not show up at the right method. How about creating an overload for the createForImportJob and createForExportJob ?